### PR TITLE
Webhook should include addtional information in the call 

### PR DESCRIPTION
**Pull request by uaArsen**
https://app.zenhub.com/workspaces/ps-development-59bbb52d21e82e515786af83/issues/elasticio/webhook-component/12

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,23 +3,26 @@
...
   "description": "elastic.io webhook component",
   "scripts": {
-    "test": "./node_modules/jasmine-node/bin/jasmine-node --test-dir spec"
+    "test": "mocha spec --recursive"
   },
   "engines": {
     "node": "6.4.0"
   },
   "author": "elastic.io GmbH",
   "license": "BSD-2-Clause",
```
Comment (uaArsen) [inline package.json]: Should be changed to: Apache License 2.0 as that what LICENSE file we have in this project

File: package.json [hunk 2]
```diff
@@ -3,23 +3,26 @@
   "version": "1.0.0",
```
Comment (ElBabbler) [inline package.json]: Should change version to 1.1.0
```suggestion
  "version": "1.1.0",
```

File: package.json [hunk 3]
```diff
@@ -3,23 +3,26 @@
...
   "description": "elastic.io webhook component",
   "scripts": {
-    "test": "./node_modules/jasmine-node/bin/jasmine-node --test-dir spec"
+    "test": "mocha spec --recursive"
```
Comment (ElBabbler) [inline package.json]: It's better to add checkstyle script run (also add eslint to the component)

File: .eslintrc.js [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+const WARN = 'warn';
+const ALWAYS = 'always';
+const NEVER = 'never';
+const OFF = 'off';
+
+module.exports = {
+    'env': {
+        es6: true,
+        node: true,
+        jasmine: true,
```
Comment (shulkaolka) [inline .eslintrc.js]: Jasmine isn't used for this component, right?

File: receive.js [hunk 5]
```diff
@@ -1,24 +1,44 @@
...
-    console.log("Received new message with id", msgId);
+    const msgId = msg.id;
+    console.log('Received new message with id', msgId);
     console.log(msg);
 
-    if (msg.query && msg.body) {
-        msg.body._query = msg.query;
+    if (msg.body) {
+        if (msg.query) {
+            msg.body._query = msg.query;
```
Comment (shulkaolka) [inline receive.js]: In the task, it wasn't necessary to redefine `msg.body._query`. Why did you decide to do this?

File: receive.js [hunk 6]
```diff
@@ -1,24 +1,44 @@
...
+        }
+
+        if (msg.headers) {
+            msg.body._headers = msg.headers;
+        }
+        if (msg.method) {
+            msg.body.__method = msg.method;
```
Comment (shulkaolka) [inline receive.js]: Why there are 2 underscores?

File: receive.js (hunk 5 above)
Comment (shanthoshp) [reply receive.js]: I didn't redefine ```msg.body._query```, it still equals ```msg.query```

File: receive.js (hunk 6 above)
Comment (shanthoshp) [reply receive.js]: Good catch, error from my end

**Comment by shanthoshp** [comment]
reopen

**Comment by shanthoshp** [comment]
For the task it does not say to write integration tests... do we want to add this as part of the story now?

**Comment by shanthoshp** [comment]
For the ```CHANGELOG``` what type of information is required?

File: package.json (hunk 3 above)
Comment (ElBabbler) [reply package.json]: Please, ensure that it's no errors left

File: lib/actions/request.js [hunk 7]
```diff
@@ -1,16 +1,17 @@
-var request = require('request');
-var qs = require('querystring');
-var messages = require('elasticio-node').messages;
-var debug = require('debug')('webhook:request');
+'use strict';
+const request = require('lib/actions/request');
```
[the remark this answers is not in the document]
Comment (shanthoshp) [reply lib/actions/request.js]: I think webstorm dynamically changed it once I moved the ```request``` file into actions. But this request should be equal to the library ```request`` and not the action

File: lib/helper/request.js [hunk 8]
```diff
@@ -1,25 +1,28 @@
...
+    let requestSettings = buildRequestSettings(method, uri, conf.secret);
     requestSettings.body = JSON.stringify(body);
     requestSettings.headers['Content-Type'] = 'application/json;charset=UTF-8';
 
+    // eslint-disable-next-line no-use-before-define
     request(requestSettings, callback.bind(this));
 };
 
-exports.get = function get(msg, conf, next) {
+exports.get = function get(msg, conf,) {
```
Comment (shulkaolka) [inline lib/helper/request.js]:
```suggestion
exports.get = function get(msg, conf) {
```

File: lib/triggers/receive.js [hunk 9]
```diff
@@ -0,0 +1,60 @@
+'use strict';
+const Q = require('q');
```
Comment (shulkaolka) [inline lib/triggers/receive.js]:
```suggestion
const Q = require('q');
const debug = require('debug')('webhook:receive');
```

File: lib/triggers/receive.js [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+
+// eslint-disable-next-line func-names
+exports.process = function (msg) {
+    const msgId = msg.id;
+    // eslint-disable-next-line no-console
+    console.log('Received new message with id', msg);
```
Comment (shulkaolka) [inline lib/triggers/receive.js]:
```suggestion
   debug('Received new message with id', msg);
```

File: lib/triggers/receive.js [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+        if (msg.url) {
+            msg.body._url = msg.url;
+        }
+
+        if (msg.additionalUrlPath) {
+            msg.body._additionalUrlPath = msg.additionalUrlPath;
+        }
+
+        // eslint-disable-next-line no-console
+        console.log('Updated body', msg.body);
```
Comment (shulkaolka) [inline lib/triggers/receive.js]:
```suggestion
        debug('Updated body', msg.body);
```

File: spec/webhook.spec.js [hunk 12]
```diff
@@ -1,326 +1,316 @@
...
+const receive = require('../lib/triggers/receive.js');
+
+describe('Test Webhook', () => {
+    afterEach(() => {
+        sinon.reset();
     });
+    const webhookReturnObj = { message: 'ok', other: 'returned' };
+    let self;
+
+    it('PUT No Auth', async () => {
```
Comment (shulkaolka) [inline spec/webhook.spec.js]:
```suggestion
    it('PUT No Auth', () => {
```

File: spec/webhook.spec.js [hunk 13]
```diff
@@ -1,326 +1,316 @@
...
-                .put('/test', {k1:'v1', k2:'v2'})
-                .matchHeader('Content-Type', 'application/json;charset=UTF-8')
-                .reply(200, webhookReturnObj, {
-                    'Content-type': 'application/json;charset=UTF-8'
-                });
-
-            runs(function() {
-                send.process.call(self, {
-                    body:{k1:'v1', k2:'v2'}
+        await new Promise((resolve) => {
```
Comment (shulkaolka) [inline spec/webhook.spec.js]:
```suggestion
        const executeAction = new Promise((resolve) => {
```

File: spec/webhook.spec.js [hunk 14]
```diff
@@ -1,326 +1,316 @@
...
                     method: 'PUT'
                 });
-            });
-
-            waitsFor(function(){
-                return self.emit.calls.length === 2;
-            }, 'Timed Out', 1000);
+        }
+        );
+        expect(nockObj.isDone());
```
Comment (shulkaolka) [inline spec/webhook.spec.js]:
```suggestion
               executeAction.then(function resolve() {
            expect(nockObj.isDone());
            expect(self.calledTwice).to.be.true;
            expect(self.args[0][1].body).to.eql(webhookReturnObj);
            expect(self.args[1][0]).to.eql('end');
        });
```

**Comment by shanthoshp** [comment]
Ready for QA - Issue was a part of sprint #20

**Pull request merged**